packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx

```diff
@@ -104,28 +108,53 @@ export const NewUserEmailForm = ({
   );
 };
 
-export function emailInputValidationAndAccountCheck(
+export async function checkAccountExists(userEmail: string) {
+  const response = await fetch(`${config.servers.auth.url}/v1/account/status`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ email: userEmail }),
+  });
+  return response.json();
+}
+
+export async function emailInputValidationAndAccountCheck(
   value: string,
   focused: boolean,
   setEmailInputState: Function,
-  getString: Function
+  getString: Function,
+  checkAccountExists: Function
 ) {
   let error = null;
   let valid = false;
+  let hasAccount = false;
   setEmailInputState(value);
   if (isEmailValid(value)) {
-    // check if we have an existing account here
     valid = true;
+    await checkAccountExists(value).then((response: { exists: boolean }) => {
+      if (response.exists) {
+        hasAccount = true;
+      }
+    });
```

Comment:
try not to mix `await` and `then`

Reply:
lol fixed, not sure what I was thinking there